### DISCUSSION TITLE
installing issue

### DISCUSSION THREAD

**Discussion by npj06** [catboost/catboost#2697] [Q&A]
I'm encountering errors when installing R packages, how can I resolve them?
R version 4.3.3 (2024-02-29)
<img width="730" alt="截屏2024-06-22 19 03 58" src="https://github.com/catboost/catboost/assets/114918453/48fec878-31a7-40a1-8837-2553cd3c4d31">

installing code:remotes::install_url('https://github.com/catboost/catboost/releases/download/v1.2.5/catboost-R-windows-x86_64-1.2.5.tgz', INSTALL_opts = c("--no-multiarch", "--no-test-load"))

**Comment by andrey-khropov** [comment]
Check your network connection. Can you download the file itself

https://github.com/catboost/catboost/releases/download/v1.2.5/catboost-R-windows-x86_64-1.2.5.tgz

using other means (for example, browser)?

**Comment by npj06** [comment]
Subject: macOS Compatibility Issue


Thank you for your help. I've tried downloading the software, but it's not compatible with my macOS system. My network is fine, as I can install other packages without issues.


Could you suggest an alternative solution?





3℃
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "catboost/catboost"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月8日(星期一) 晚上7:01
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [catboost/catboost] installing issue (Discussion #2697)





 
Check your network connection. Can you download the file itself
 
https://github.com/catboost/catboost/releases/download/v1.2.5/catboost-R-windows-x86_64-1.2.5.tgz
 
using other means (for example, browser)?

**Reply by andrey-khropov** [reply]
> I've tried downloading the software, but it's not compatible with my macOS system. 

In the original message you've downloaded the Windows package. Is this a different issue or you tried it on macOS machine? 

For macOS you should use "https://github.com/catboost/catboost/releases/download/v1.2.5/catboost-R-darwin-universal2-1.2.5.tgz" .

**Comment by npj06** [comment]
still not working




3℃
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "catboost/catboost"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月8日(星期一) 晚上7:45
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [catboost/catboost] installing issue (Discussion #2697)





  
I've tried downloading the software, but it's not compatible with my macOS system.
  
In the original message you've downloaded the Windows package. Is this a different issue or you tried it on macOS machine?
 
For macOS you should use "https://github.com/catboost/catboost/releases/download/v1.2.5/catboost-R-darwin-universal2-1.2.5.tgz" .

**Comment by andrey-khropov** [comment]
Try adding '--no-staged-install' option: https://github.com/catboost/catboost/issues/1985#issuecomment-1110131518

**Comment by mrtclb2301-cyber** [comment]
I treid to dowload as above instruct,ons but I am receiving this message: Error in .rs.downloadFile(url = url, destfile = path, method = method,  : 
  download from 'https://api.github.com/repos/catboost/catboost/tarball/HEAD' failed

**Reply by andrey-khropov** [reply]
> I treid to dowload as above instruct,ons but I am receiving this message: Error in .rs.downloadFile(url = url, destfile = path, method = method, : download from 'https://api.github.com/repos/catboost/catboost/tarball/HEAD' failed

Can you provide more details?

- OS version
- CatBoost version
- R version + how R has been installed
- The exact command that you used to run CatBoost R package install
